Sources/SwiftProtobuf/FieldNameMap.swift

```diff
@@ -8,6 +8,82 @@
 //
 // -----------------------------------------------------------------------------
 
+/// Proto field names are always ASCII, so we can improve the
+/// performance of some of our parsing by storing them as
+/// arrays of UInt8 to avoid the overhead of creating and
+/// comparing Unicode Strings.
```

Comment:
I just pushed a commit with this change.  A few interesting points:

 * I did not have to make AsciiFieldName public:  It gets initialized directly from the string literals, so we just need to say we want StaticString there and the type inference lines it all up for us.

* I did have to change the code generation to provide a way for TextDecoder to notice when it's encoding extensions -- the former code relied on the extension to be able to create `FieldNameMap.Names` objects at run-time, which we can't do with `StaticStrings`.  So I added a (long overdue) path for encoders to actually know when they are encoding extensions.